base/rational.jl

```diff
@@ -145,6 +145,8 @@ function (::Type{T})(x::Rational{S}) where T<:AbstractFloat where S
     P = promote_type(T,S)
     convert(T, convert(P,x.num)/convert(P,x.den))::T
 end
+Float16(x::Rational{<:Union{Int16,Int32,Int64,Int128,UInt16,UInt32,UInt64,UInt128}}) =
```

Comment:
`Int16` is not necessary here; see this [comment](https://github.com/JuliaLang/julia/issues/52394#issuecomment-1853158388).

However(!) `UInt128` can still produce spurious overflow since it can overflow `Float32`.
The easy fix is to add
```julia
Float32(x::Rational{UInt128}) = Float32(Float64(x))
```